src/framework/components/light/component.js

```diff
@@ -178,6 +54,839 @@ class LightComponent extends Component {
         this._cookieMatrix = null;
     }
 
+    // TODO: Remove this override in upgrading component
+    /**
+     * @type {import('./data.js').LightComponentData}
+     */
+    get data() {
+        const record = this.system.store[this.entity.getGuid()];
+        return record ? record.data : null;
+    }
+
+    /**
+     * @type {boolean}
+     */
+    set enabled(arg) {
+        this._setValue('enabled', arg, function (newValue, oldValue) {
+            this.onSetEnabled(null, oldValue, newValue);
+        });
+    }
+
+    get enabled() {
+        return this.data.enabled;
+    }
+
+    /**
+     * @type {import('../../../scene/light.js').Light}
+     * @ignore
+     */
+    set light(arg) {
+        this._setValue('light', arg);
+    }
+
+    get light() {
+        return this.data.light;
+    }
+
+    /**
+     * The type of light. Can be:
+     *
+     * - "directional": A light that is infinitely far away and lights the entire scene from one
+     * direction.
```

Comment:
ideally instead of string types, we use the exported enums here

Reply:
I will prob go back and change fixup the types when upgrading the class again to move the majority of properties to the component instead of the data class